ENIGMAsystem/SHELL/Universal_System/Resources/asset_index.cpp

```diff
@@ -0,0 +1,37 @@
+#include "asset_index.h"
+
+using enigma::assetMap;
+using enigma::assetTypes;
+
+namespace enigma_user {
+
+int asset_get_index(std::string resource, AssetType type) {
+ if (type == asset_any) {
+  for (const AssetType t : assetTypes) {
+    if (assetMap.count(t) == 0) continue;
+    const auto& m = assetMap.at(t);
+    auto it = m.find(resource);
+    if (it != m.end()) return it->second;
+    } 
```

Comment:
Fix the indentation here.

```suggestion
  if (type == asset_any) {
    for (const AssetType t : assetTypes) {
      if (assetMap.count(t) == 0) continue;
      const auto& m = assetMap.at(t);
      auto it = m.find(resource);
      if (it != m.end()) return it->second;
    } 
```